data/src/main/java/com/microsoft/azure/kusto/data/ClientRequestProperties.java

```diff
@@ -186,19 +194,29 @@ private long parseTimeoutFromTimespanString(String str) throws ParseException {
                 timespanWithoutDays += matcher.group(i);
             }
         }
-        millis += TimeUnit.NANOSECONDS.toMillis(LocalTime.parse(timespanWithoutDays).toNanoOfDay());
-        return millis;
+
+        return TimeUnit.NANOSECONDS.toMillis(LocalTime.parse(timespanWithoutDays).toNanoOfDay());
     }
 
     /**
      * Sets the amount of time a query may execute on the service before it times out.
-     * @param timeoutInMs number of milliseconds before timeout. Value must be between 1 minute and 1 hour.
+     * @param timeoutInMs number of milliseconds before timeout.
+     *                    Value must be between 1 minute and 1 hour, and so value below the minimum or above the maximum will be adjusted accordingly.
      */
     public void setTimeoutInMilliSec(Long timeoutInMs) {
-        if (timeoutInMs < 60000 || timeoutInMs > MAX_TIMEOUT_MS) {
-            throw new IllegalArgumentException(String.format("Timeout must be between 60,000 and %,d milliseconds", MAX_TIMEOUT_MS));
+        options.put(OPTION_SERVER_TIMEOUT, calculateTimeoutPerServiceLogic(timeoutInMs));
```

Comment:
This change was the impetus for this PR, as @yogilad said it's better to not break users' existing usages.